tests/filecheck/dialects/comb/comb_ops.mlir

```diff
@@ -0,0 +1,69 @@
+  %lhsindex, %rhsindex = "test.op"() : () -> (index, index)
+  %lhsf32, %rhsf32 = "test.op"() : () -> (f32, f32)
+  %lhsf64, %rhsf64 = "test.op"() : () -> (f64, f64)
+  %lhsvec, %rhsvec = "test.op"() : () -> (vector<4xf32>, vector<4xf32>)
+
+  %add = "comb.add"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK: %add = "comb.add"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %mul = "comb.mul"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %mul = "comb.mul"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %divu = "comb.divu"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %divu = "comb.divu"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %divs = "comb.divs"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %divs = "comb.divs"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %modu = "comb.modu"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %modu = "comb.modu"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %mods = "comb.mods"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %mods = "comb.mods"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %shl = "comb.shl"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %shl = "comb.shl"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %shru = "comb.shru"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %shru = "comb.shru"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %shrs = "comb.shrs"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %shrs = "comb.shrs"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %sub = "comb.sub"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %sub = "comb.sub"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %and = "comb.and"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %and = "comb.and"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %or = "comb.or"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %or = "comb.or"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %xor = "comb.xor"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %xor = "comb.xor"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %parity = "comb.parity"(%lhsi1) : (i1) -> i1
+  // CHECK-NEXT: %parity = "comb.parity"(%lhsi1) : (i1) -> i1
+
+  %extract = "comb.extract"(%lhsi32) {"low_bit" = 1 : i32} : (i32) -> i32
+  // CHECK-NEXT: %extract = "comb.extract"(%lhsi32) {"low_bit" = 1 : i32} : (i32) -> i32
+
+  %concat = "comb.concat"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+  // CHECK-NEXT: %concat = "comb.concat"(%lhsi32, %rhsi32) : (i32, i32) -> i32
+
+  %replicate = "comb.replicate"(%lhsi32) : (i32) -> i32
+  // CHECK-NEXT: %replicate = "comb.replicate"(%lhsi32) : (i32) -> i32
+
+  %mux = "comb.mux"(%lhsi1, %lhsi32, %rhsi32) : (i1, i32, i32) -> i32
+  // CHECK-NEXT: %mux = "comb.mux"(%lhsi1, %lhsi32, %rhsi32) : (i1, i32, i32) -> i32
+
+  %icmp = "comb.icmp"(%lhsi1, %rhsi1) {"predicate" = 2 : index, "two_state"} : (i1, i1) -> i1
+  // CHECK-NEXT: %icmp = "comb.icmp"(%lhsi1, %rhsi1) {"predicate" = 2 : index, "two_state"} : (i1, i1) -> i1
+  }) : () -> ()
+  
```

Comment:
```suggestion

```
some stray trailing whitespace

Reply:
@dshaaban01 I believe the permanent solution to avoid spurious insertions in the future in VSCode is the setting

```json
"files.trimTrailingWhitespace": true
```

Searching for "trailing" in preferences should find the relevant UI checkbox.

Reply:
Hmm, I'm actually confused why this isn't checked by pre-commit, maybe I made a mistake there?

Reply:
@math-fehr hmm, yeah that was my initial thought, but I didn't have time to look into it and `black`.